samplers/jaegerremote/sampler_remote_options.go

```diff
@@ -0,0 +1,130 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerremote
+
+import (
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type config struct {
+	sampler                 trace.Sampler
+	samplingServerURL       string
+	samplingRefreshInterval time.Duration
+	samplingFetcher         samplingStrategyFetcher
+	samplingParser          samplingStrategyParser
+	updaters                []samplerUpdater
+	posParams               perOperationSamplerParams
+}
+
+// newConfig returns an appropriately configured config.
+func newConfig(options ...Option) config {
+	c := config{
+		sampler:                 newProbabilisticSampler(0.001),
+		samplingServerURL:       defaultSamplingServerURL,
+		samplingRefreshInterval: defaultSamplingRefreshInterval,
+		samplingFetcher:         newHTTPSamplingStrategyFetcher(defaultSamplingServerURL),
+		samplingParser:          new(samplingStrategyParserImpl),
+		updaters: []samplerUpdater{
+			&perOperationSamplerUpdater{
+				MaxOperations:            defaultSamplingMaxOperations,
+				OperationNameLateBinding: defaultSamplingOperationNameLateBinding,
+			},
+			new(probabilisticSamplerUpdater),
+			new(rateLimitingSamplerUpdater),
+		},
+	}
+	for _, option := range options {
+		option.apply(&c)
+	}
+
+	return c
+}
+
+type Option interface {
+	apply(*config)
+}
+
+type optionFunc func(*config)
+
+func (fn optionFunc) apply(c *config) {
+	fn(c)
+}
+
+// WithInitialSampler creates a Option that sets the initial sampler
+// to use before a remote sampler is created and used.
+func WithInitialSampler(sampler trace.Sampler) Option {
+	return optionFunc(func(c *config) {
+		c.sampler = sampler
+	})
+}
+
+// WithSamplingServerURL creates a Option that sets the sampling server url
+// of the local agent that contains the sampling strategies.
+func WithSamplingServerURL(samplingServerURL string) Option {
+	return optionFunc(func(c *config) {
+		c.samplingServerURL = samplingServerURL
+		// The default port of jaeger agent is 5778, but there are other ports specified by the user, so the sampling address and fetch address are strongly bound
+		c.samplingFetcher = newHTTPSamplingStrategyFetcher(samplingServerURL)
+	})
+}
+
+// WithMaxOperations creates a Option that sets the maximum number of
+// operations the sampler will keep track of.
+func WithMaxOperations(maxOperations int) Option {
+	return optionFunc(func(c *config) {
+		c.posParams.MaxOperations = maxOperations
+	})
+}
+
+// WithOperationNameLateBinding creates a Option that sets the respective
+// field in the perOperationSamplerParams.
+func WithOperationNameLateBinding(enable bool) Option {
+	return optionFunc(func(c *config) {
+		c.posParams.OperationNameLateBinding = enable
+	})
+}
```

Comment:
These options appear to be ineffective.

Reply:
> These options appear to be ineffective.
This is used in sapmler_ remote. go  L242

Reply:
Do you mean L252?  Even then, no, those are not the values that these options affect.